core-war/src/main/webapp/util/javaScript/vuejs/components/comments/silverpeas-comments.js

```diff
@@ -0,0 +1,257 @@
+                  return sp.ajaxRequest(this.url + '/' + commentWithUpdatedText.id)
+                      .byPutMethod()
+                      .sendAndPromiseJsonResponse(commentWithUpdatedText)
+                      .then(function(updatedComment) {
+                        this.emitChange('update', updatedComment);
+                        return this.loadComments();
+                      }.bind(this));
+                }
+                return false;
+              }.bind(this)
+            });
+          },
+          /**
+           * Deletes the specified existing comment on the given Silverpeas resource for
+           * the current user.
+           * @param comment {Object} the comment to delete.
+           */
+          deleteComment: function(comment) {
+            sp.popup.confirm(this.messages.commentDeletionConfirmation, function() {
+              return sp.ajaxRequest(this.url + '/' + comment.id)
+                  .byDeleteMethod()
+                  .send()
+                  .then(function() {
+                    this.emitChange('deletion', comment);
+                    return this.loadComments();
+                  }.bind(this));
+            }.bind(this));
+          },
+          emitChange: function(type, comments) {
+            this.$emit(type, Array.isArray(comments) ? comments : Array(comments));
+          },
+        },
+        computed : {
+          infoNbOfComments: function() {
+            let msg;
+            if (this.comments.length === 0) {
+              msg = this.messages.textNoComment;
+            } else if (this.comments === 1) {
+              msg = '1' + this.messages.textComment;
+            } else {
+              msg = this.comments.length + ' ' + this.messages.textComments;
+            }
+            return msg;
+          }
+        }
+      }));
+
+  Vue.component('silverpeas-comment-edition',
+      templateRepository.get('comment-edition', {
+        model : {
+          prop : 'value',
+          event : 'input'
+        },
+        props : {
+          currentUser: {
+            'type': Object,
+            'mandatory': true
+          },
+          value : {
+            'type' : String
+          }
+        },
+        mounted : function() {
+          let $textarea = this.$el.querySelector('#comment-edition-text');
+          jQuery($textarea).autoResize();
+        },
+        data: function() {
+          return {
+            comment: ''
+          }
+        }
+      }));
+
+  Vue.component('silverpeas-comment',
+      templateRepository.get('comment', {
+        props: {
+          currentUser: {
+            'type': Object,
+            'mandatory': true
+          },
+          comment: {
+            'type': Object,
+            'mandatory': true
+          },
+          readonly: {
+            'type': Boolean,
+            'default': false
+          }
+        },
+        mounted: function() {
+          activateUserZoom();
```

Comment:
Maybe computed `displayUserZoom` variable could be used here to avoid calling the user zoom initialization?

Reply:
I don't understand what you mean here.
The computed data `displayUserZoom` is used to put or not the CSS class `userToZoom` that is used by the UserZoom widget when it is activated to attach the widget to such elements. And, to ensure the activation of the UserZoom is well performed once the VueJS components are mounted on the DOM, a call to `activateUserZoom()` is done here. Another way to do that is to perform an explicit call directly to the HTML element to which a UserZoom widget has to be attached. But I preferred the former approach.